odl/solvers/functional/default_functionals.py

```diff
@@ -291,22 +291,8 @@ def __init__(self):
 
             def _call(self, x):
                 """Return ``self(x)``."""
-                p = functional.pointwise_norm.exponent
-
-                if functional.pointwise_norm.exponent == 1:
-                    result = np.abs(x)
-                    np.divide(x, result, out=result, where=result != 0)
-                    return result
-                elif functional.pointwise_norm.exponent == 2:
-                    result = functional.pointwise_norm(x)
-                    np.divide(x, result, out=result, where=result != 0)
-                    return result
-                else:
-                    dividend = np.power(np.abs(x), p - 2) * x
-                    divisor = np.power(functional.pointwise_norm(x), p - 1)
-                    np.divide(dividend, divisor, out=divisor,
-                              where=divisor != 0)
-                    return divisor
+                return functional.pointwise_norm.derivative(x).adjoint(
+                    np.sign(functional.pointwise_norm(x)))
```

Comment:
Nice reduction! Can we also try to optimize things a bit here? Not much, only add `out` to the signature and do
```python
pwnorm_x = functional.pointwise_norm(x)
pwnorm_x.ufuncs.sign(out=pwnorm_x)
functional.pointwise_norm.derivative(x).adjoint(pwnorm_x, out=out)
return out
```